media_utils/youtube_puppeteer.py

```diff
@@ -235,12 +241,9 @@ def ad_ended(self):
         return self.ad_state == self._yt_player_state['ENDED']
 
     def process_ad(self):
-        if self.attempt_ad_skip():
-            return
-        ad_duration = self.search_ad_duration()
-        if not ad_duration:
+        if self.attempt_ad_skip() or self.ad_inactive:
             return
-        ad_timeout = ad_duration + 5
+        ad_timeout = (self.search_ad_duration() or 30) + 5
```

Comment:
Well, on second thought, this value is specific to ads, not video start-up time or the video as a whole. If someone initializes a video with a short timeout, say, I don't want the ad_timeout to change.